minimailgun/api.py

```diff
@@ -0,0 +1,118 @@
+"""
+Implement the Flask API
+"""
+
+import os
+import itertools
+import traceback
+from functools import wraps
+
+from flask import Flask, Blueprint, jsonify, request, abort, current_app
+from werkzeug.exceptions import default_exceptions, HTTPException
+
+from minimailgun.store import store, MailLookupError, UnablToAddMessageError
+from minimailgun.tasks import handle_new_message
+
+
+class validate_mimetype(object):
```

Comment:
When I spoke about the decorator, I meant the decorator that would combine the validation with configuring a route so you'd have just a single "custom-made" decorator that you could apply to all handlers.

```python
def api(path, methods, ...):
    # decorator that does validation and registers a route
    ...

@api("/mail", methods=["POST"])
def create_mail():
    ...
```

That is what we have in our API, but what you implemented also does the job. JFYI.